contributors/design-proposals/security-threat-model.md

```diff
@@ -0,0 +1,89 @@
+# Security Threat Model
+
+As a cluster manager Kubernetes is expected to defend against a wide variety of threats. This document
+attempts to define the high level actors, potential threats, and our primary layers of defense.
+
+
+## Brief architectural description.
+
+Kubernetes is a cluster manager that runs application workloads inside of containers.  Workloads run as
+**pods** on **nodes** which communicate with the **api-server** to retrieve the description and list of
+pods. Pods may communicate with each other over the network, or contact virtual proxies known as
+**services** from each node at a unique IP or DNS address. Traffic to pods from outside the cluster
+may flow via an **ingress** router/proxy. Pods may have additional resources attached at runtime,
+including persistent volumes, credentials that let them act against the API, secrets used for contacting
+other systems, or configuration files for their processes.
+
+Users and administrators access the apiserver to deploy applications, control access, or make
+changes to existing applications. **Controllers** are automated agents of the system that use the
+definitions in the api-server to ensure the desired state matches reality. Controllers act on-behalf-of
+users to make these changes and have access privileges regular users will not.
+
+
+## Actors
+
+The primary actors and their roles on the system are:
+
+1. Users deploy applications onto the platform via the APIs related to application definition.
+   1. Some users are highly limited in the actions they may take, such as only creating a subset of available resources and being unable to edit.
+2. Administrators manage security, allocate resources, and conduct maintenance on cluster components.
+   1. Organization admins control a subset of resources on the cluster, but are not privileged to act on the whole cluster.
+   2. Infrastucture admins manage both the cluster and the underlying systems that run the cluster.
+3. Each node offers resources to the cluster and accepts pods that have been scheduled on it.
+4. Scheduler controllers assign pods to nodes based on workload definitions and must know details of both to effectively schedule.
+5. Workload controllers create pods that have certain lifecycle rules.
+6. Network providers use information about the topology of the cluster (physical and logical) to give pods IP address and control their access.
+7. Secret controllers create or update secrets for use by users and applications.
+8. Federations need to run workloads on behalf of users on individual clusters.
+9. Other integrations may need access to read or write definitions on some or all of the namespaces in the cluster.
+
+
+## Security boundaries
+
+Several important security boundaries exist in Kubernetes that reduce the possibility of interference or exploit by consumers.
```

Comment:
I assume consumers is the superset of all the actors above. To reduce confusion I'd drop it and just use:

```
Several important security boundaries exist in Kubernetes.
```